sox: fix device name length

genevera opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Applies patch to fix problem with device name length being inaccurate in sox on macOS.

carlocab commented on Formula/sox.rb:
```diff
@@ -29,10 +29,16 @@ class Sox < Formula
     sha256 "83af02f2aa2b746bb7225872cab29a253264be49db0ecebb12f841562d9a2923"
   end
 
+  # Applies Eric Wong's patch to fix device name length in MacOS
```
Can you add when we might be able to remove this patch to the comment?

carlocab commented on Formula/sox.rb:
```diff
@@ -29,10 +29,16 @@ class Sox < Formula
+  patch do
+    url "https://80x24.org/sox.git/patch?id=bf2afa54a7dec"
+    sha256 "0cebb3d4c926a2cf0a506d2cd62576c29308baa307df36fddf7c6ae4b48df8e7"
+  end
+
   def install
     system "./configure", "--disable-debug",
-                          "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
+      "--disable-dependency-tracking",
+      "--prefix=#{prefix}"
```
Please don't change this.

carlocab reviewed (commented):
Thanks for the PR.

Please use the preferred commit-message style for homebrew/core. We put the name of the formula first in commit-message headings.

For new formulae:

> At Homebrew, we like to put the name of the formula up front like so: `foobar 7.3 (new formula)`.

For existing formulae:

> The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.

Refer to the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit) for more details. Also, when making further changes to your pull request, use the following guidelines to make sure that @BrewTestBot can merge your commits:

> - One formula per commit; one commit per formula.
> - Keep merge commits out of the pull request.

genevera commented:
@carlocab Thank you for the review!

I added info to the code comment, and fixed the spacing in that block.  My vim auto-formatter edited that.  Is there an .editorconfig file or other syntax file I can use with ruby-beautify or rubocop to ensure the spacing is preserved?

genevera commented:
Hi @carlocab - just following up on this since it's been about a week. I hope you'll have time to have a look at the requested changes that I've committed and lmk if anything is missing still.

carlocab reviewed (approved):
LGTM; thanks!
